### ISSUE TITLE
Fix insane data paths

### ISSUE THREAD

**Issue by ghost** [mumble-voip/mumble#1919]
Since the data directory is being migrated with 1.3 anyway, you might as well get it right this time.

Instead of `~/.local/share/data/Mumble/Mumble/` or `~/.local/share/Mumble/Mumble/`,
put data in `~/.local/share/Mumble/`.

Also, why are the database and journal hidden files?
Make them

```
~/.local/share/Mumble/mumble.sqlite
~/.local/share/Mumble/mumble.sqlite-journal
```

Sanity thanks you.
Sorry if this is a duplicate or if there's some actual weird reason for this.

**Comment by mkrautz** [comment]
They're not really intentionally being migrated in 1.3.0.

If they are, maybe we should just keep the originals instead?

I agree that what you posted there seems extreme. It's the output of using Qt's QStandardPaths API. Maybe we're misusing it though, I don't recall.

**Comment by rolf1304** [comment]
You're not misusing it. The Qt5 version of Mumble uses QDesktopServices instead of QStandardPaths, which has (rightfully) dropped the `data/` directory.
From the Qt documentation:
>Note: when porting QDesktopServices::DataLocation to QStandardPaths::DataLocation, a different path will be returned.
QDesktopServices::DataLocation was GenericDataLocation + "/data/organization/application", while QStandardPaths::DataLocation is GenericDataLocation + "/organization/application".

Which also explains the needlessly double `Mumble/Mumble/` directory.
That means you have three justifiable options:
* Like it is now, use QDesktopServices::DataLocation, which will break compatibility with older Mumble versions.
* Don't use the DataLocation, but instead use GenericDataLocation + "Mumble" in Mumble 1.3. This also breaks compatibility, but is sane and consistent.
* Revert to using the silly `data/Mumble/Mumble/` directory, which won't break things (except for those people who are already using Mumble 1.3)

**Comment by Krzmbrzl** [comment]
@davidebeatrici isn't Mumble simply letting Qt figure out where to store stuff? If so, this issue can be closed as it is Qt's fault and not ours...

**Comment by davidebeatrici** [comment]
`QStandardPaths::DataLocation` generates a path with the following format: `~/.local/share/<Organization>/<Application>`.

Since we don't have an official organization name, we are using `Mumble`:
https://github.com/mumble-voip/mumble/blob/72e1f47b7b13d2ffd2a98cb2fd79c10674376698/src/mumble/main.cpp#L102-L103

I agree that the path is not the best, because the root (organization) folder has the same name as the application one.

I don't exclude that in future we may have more than one directory in the organization one, thus giving the `Mumble` subfolder more sense. However, I would expect it to be called `Client`. In fact, @Kissaki recently proposed to consistently use "client" and "server" everywhere instead of "Mumble" and "Murmur".

Also, we're currently storing the server configuration and database files in `~/.murmurd`, while they probably should be in a "Server" folder inside the organization one.

**Comment by ZeroAbility** [comment]
Was this resolved with 1.4?

**Comment by Krzmbrzl** [comment]
No and we can't resolve it without bypassing Qt, which would require platform dependent code. This does not seem to be worth it.

The Mumble/Mumble path pops into existence because Qt also requires an organization name, for which we also chose "Mumble" (same as application name). Thus the duplication.
